source/simulator/assembly.cc

```diff
@@ -193,7 +193,8 @@ namespace aspect
             assemblers->advection_system_assembler_on_face_properties[0].need_face_finite_element_evaluation = true;
           }
 
-        if (i > 0 && parameters.use_discontinuous_composition_discretization)
+        if (i > 0 && parameters.use_discontinuous_composition_discretization
+            && parameters.compositional_field_methods[i-1] == Parameters<dim>::AdvectionFieldMethod::fem_field)
```

Comment:
Independent of the tests, have you checked if you may also need the face assemblers for `fem_melt_field` and `fem_darcy_field` and potentially also `volume_of_fluid` ? (I dont know if those fields use the face assemblers as well). All possible values for the advection field methods can be found in parameters.h:134.